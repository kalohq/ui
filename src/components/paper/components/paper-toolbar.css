.root {
  border-bottom: 1px solid var(--grey300);
}