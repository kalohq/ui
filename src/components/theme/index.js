/* @flow */
export {default} from './theme';
