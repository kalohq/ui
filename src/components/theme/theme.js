/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
