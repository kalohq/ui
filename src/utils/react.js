import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
