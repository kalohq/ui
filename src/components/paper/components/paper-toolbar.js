/* @flow */
import * as React from 'react';

import {Box} from '../../layout';

import styles from './paper-toolbar.css';

/**
 * @summary A basic toolbar container to display at the top of paper
 */

export function PaperToolbar(props: {children?: React.Node}) {
  const {children} = props;

  return (
    <Box className={styles.root} padding={[0, 25]}>
      {children}
    </Box>
  );
}
