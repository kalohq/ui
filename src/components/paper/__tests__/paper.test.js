/* @flow */
/* eslint-env jest */
import * as React from 'react';
import * as emotion from 'emotion';
import {createSerializer} from 'jest-emotion';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';

import Paper from 'components/paper';
import {PaperToolbar} from '../components/paper-toolbar.react';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('components/paper', () => {
  describe('Paper', () => {
    const defaultProps = {};
    const create = (props = {spacing: true}) =>
      renderer
        .create(
          <ThemeProvider theme={theme}>
            <Paper {...defaultProps} {...props}>
              Paper Contents
            </Paper>
          </ThemeProvider>
        )
        .toJSON();

    test('should render shallow component ok', () => {
      const element = create();
      expect(element).toMatchSnapshot();
    });

    test('should render a component with an elevation of 3', () => {
      const element = create({
        elevation: 3,
      });
      expect(element).toMatchSnapshot();
    });
  });

  describe('PaperToolbar', () => {
    test('should render shallow component ok', () => {
      const element = renderer.create(
        <ThemeProvider theme={theme}>
          <PaperToolbar>
            <Paper>Toolbar child contents</Paper>
          </PaperToolbar>
        </ThemeProvider>
      );
      expect(element).toMatchSnapshot();
    });
  });
});
