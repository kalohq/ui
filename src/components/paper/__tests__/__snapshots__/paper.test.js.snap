// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paper Paper should render a component with an elevation of 3 1`] = `
.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  background: white;
  box-sizing: content-box;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  border: 1px solid rgb(238,244,250);
  border-radius: 5px;
  box-shadow: 0 10px 25px rgba(140,140,140,0.14);
}

.emotion-1:hover {
  box-shadow: none;
}

<div
  className="emotion-0 emotion-1"
  onClick={undefined}
  style={Object {}}
>
  Paper Contents
</div>
`;

exports[`components/paper Paper should render shallow component ok 1`] = `
.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  background: white;
  box-sizing: content-box;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  border: 1px solid rgb(238,244,250);
  border-radius: 5px;
  box-shadow: none;
}

.emotion-1:hover {
  box-shadow: none;
}

<div
  className="emotion-0 emotion-1"
  onClick={undefined}
  spacing={true}
  style={Object {}}
>
  Paper Contents
</div>
`;

exports[`components/paper PaperToolbar should render shallow component ok 1`] = `
.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  background: white;
  box-sizing: content-box;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  border: 1px solid rgb(238,244,250);
  border-radius: 5px;
  box-shadow: none;
}

.emotion-1:hover {
  box-shadow: none;
}

.emotion-3 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px 25px;
  border-bottom: 1px solid rgb(238,244,250);
}

<div
  className="emotion-2 emotion-3"
  style={Object {}}
>
  <div
    className="emotion-0 emotion-1"
    onClick={undefined}
    style={Object {}}
  >
    Toolbar child contents
  </div>
</div>
`;
